Composer/packages/client/src/utils/auth.ts

```diff
@@ -327,27 +327,19 @@ export function getAccessTokenUrl(options: { clientId: string; redirectUrl: stri
   return url;
 }
 
-export function isShowAuthDialog(needGraph: boolean): boolean {
+export function userShouldProvideTokens(): boolean {
   if (isElectron()) {
     return false;
-  } else if (!(authConfig.clientId && authConfig.redirectUrl && authConfig.tenantId)) {
-    return isTokenExpired(getTokenFromCache('accessToken'))
-      ? true
-      : needGraph && isTokenExpired(getTokenFromCache('graphToken'))
-      ? true
-      : false;
-  } else {
-    return false;
-  }
+  } else return !(authConfig.clientId && authConfig.redirectUrl && authConfig.tenantId);
 }
 
-export function userShouldProvideTokens(): boolean {
-  if (isElectron()) {
-    return false;
-  } else if (authConfig.clientId && authConfig.redirectUrl && authConfig.tenantId) {
+export function isShowAuthDialog(needGraph: boolean): boolean {
+  if (userShouldProvideTokens())
+    return (
+      (needGraph && isTokenExpired(getTokenFromCache('graphToken'))) || isTokenExpired(getTokenFromCache('accessToken'))
+    );
+  else {
     return false;
-  } else {
-    return true;
```

Comment:
Sorry about the mess here; Git's diff viewer got confused by my swapping the position of these two functions. This isn't as big a change as it looks like.